GVRf/Framework/framework/src/main/jni/engine/renderer/vulkan_renderer.cpp

```diff
@@ -120,8 +120,11 @@ RenderTexture* VulkanRenderer::createRenderTexture(int width, int height, int sa
                                                    int jcolor_format, int jdepth_format, bool resolve_depth,
                                                    const TextureParameters* texture_parameters, int number_views, bool monoscopic)
 {
-    if(monoscopic)
-        return new VkRenderTextureOnScreen(width, height, sample_count);
+    if(monoscopic) {
+        VkRenderTextureOnScreen * tex = new VkRenderTextureOnScreen(width, height, sample_count);
+        mOnScreenTextures.push_back(tex);
```

Comment:
Let's find a way to avoid expanding upon the monoscopic-specific hack we already have in here. The MonoscopicViewManager already tracks the renderTextures via the mRenderTarget array. Why can't we recreate/reset the old renderTextures from the Java side from surfaceCreated?

Reply:
working on this. Instead of destroying the renderTargets on termination of draw thread, as we are doing right now, we could possibly retain them and just switch the swapChain image handles in recreateSwapChain.